src/stacktraces.cpp

```diff
@@ -298,10 +333,122 @@ static std::vector<stackframe_info> GetStackFrameInfos(const std::vector<uintptr
     return infos;
 }
 
-static std::string GetStackFrameInfosStr(const std::vector<stackframe_info>& sis, size_t spaces = 2)
+struct crash_info_header
+{
+    std::string magic;
+    uint16_t version;
+    std::string exeFileName;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITE(magic);
+        READWRITE(version);
+        READWRITE(exeFileName);
+    }
+};
+
+struct crash_info
+{
+    std::string crashDescription;
+    std::vector<uint64_t> stackframes;
+    std::vector<stackframe_info> stackframeInfos;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
+    {
+        READWRITE(crashDescription);
+        READWRITE(stackframes);
+        READWRITE(stackframeInfos);
+    }
+
+    void ConvertAddresses(int64_t offset)
+    {
+        for (auto& sf : stackframes) {
+            sf += offset;
+        }
+        for (auto& sfi : stackframeInfos) {
+            sfi.pc += offset;
+        }
+    }
+};
+
+static std::string GetCrashInfoStrNoDebugInfo(crash_info ci)
+{
+    static uint64_t basePtr = GetBaseAddress();
+
+    CDataStream ds(SER_DISK, 0);
+
+    crash_info_header hdr;
+    hdr.magic = "DashCrashInfo";
+    hdr.version = 1;
+    hdr.exeFileName = g_exeFileBaseName;
+    ds << hdr;
+
+    ci.ConvertAddresses(-(int64_t)basePtr);
+    ds << ci;
+
+    auto ciStr = EncodeBase32((const unsigned char*)ds.data(), ds.size());
+    return strprintf("No debug information available for stacktrace. You should download debug information from Github and then run:\n"
```

Comment:
I would generalize this a bit:
```suggestion
    return strprintf("No debug information available for stacktrace. You should add debug information and then run:\n"
```